### DISCUSSION TITLE
Confused about Content Delivery in the self-hosted free version

### DISCUSSION THREAD

**Discussion by empz** [tolgee/tolgee-platform#2478] [Q&A] [answered]
From the pricing page it looks like Content Delivery is available in the Free and Self-hosted option:
![image](https://github.com/user-attachments/assets/c9a9afac-e13f-4ce8-949a-110e9317f924)

It looks like I cannot use my own S3 unless I upgrade to Business. But does that mean I can still use the storage provided by Tolgee?

My instance reported that Content Delivery was not configured.
![image](https://github.com/user-attachments/assets/0dcd18f4-a8a3-4d0f-9a37-916ad9ddbf2b)

So I added the following environment variable
```
TOLGEE_CONTENT_DELIVERY_PUBLIC_URL_PREFIX="mycompany"
```

Now the instance allows me to create a Content Delivery.
![image](https://github.com/user-attachments/assets/c5b4e82d-f087-476a-9249-942859f5cc25)

![image](https://github.com/user-attachments/assets/1337544a-1b91-444d-9009-68f9686c8a57)

But when I hit Save, this happens:
![image](https://github.com/user-attachments/assets/204173b4-de69-4ff9-b26f-9e427e9c6563)

The instance logs shows the following error:
```
ERROR 340 --- [nio-8080-exec-4] io.tolgee.ExceptionHandlers : java.lang.RuntimeException: Exactly one content storage must be set
```

So, is it possible to use Content Delivery with the Free / Self-hoted option? If so, how?

**Comment by JanCizmar** [comment, marked as answer]
Hello! You have to configure a storage for the content delivery. I will fix the docs.

**Reply by JanCizmar** [reply]
Here is the related issue: https://github.com/tolgee/tolgee-platform/pull/2481
